docs/tutorials/initializedirtyimage.py

```diff
@@ -0,0 +1,190 @@
+
+# +
+# %%time
+
+
+loss_tracker = []
+for iteration in range(50):
+
+    optimizer.zero_grad()
+
+    rml.forward()
+
+    sky_cube = rml.icube.sky_cube
+
+    # lossfunc = torch.nn.MSELoss(reduction='sum')  #the MSELoss calculates mean squared error (squared L2 norm), so we take sqrt of it
+    # loss = (lossfunc(sky_cube,dirty_image))**.5 #https://pytorch.org/docs/stable/generated/torch.nn.MSELoss.html
+
+    loss = l2norm(sky_cube, dirty_image)
+
+    loss_tracker.append(loss.item())
+    loss.backward()
+    optimizer.step()
+# torch.save(rml, "dirty_image_model.pt")
+torch.save(rml.state_dict(), "dirty_image_model.pt")
+# -
+
+# We also can save the state of the model or the entire model itself. The torch.save(model.state_dict(), PATH) will just save the state.dict() which are the model's learned parameters. The alternative will save the entire model.
+# https://pytorch.org/tutorials/beginner/saving_loading_models.html
+
+# ### Results
+
+fig, ax = plt.subplots(nrows=1)
+ax.plot(loss_tracker)
+ax.set_xlabel("iteration")
+ax.set_ylabel("loss")
+
+# The results are seen here with the state_dict() having changed and the image now more closely resembling the dirty image, though the dirty image contains negative values which this does not so they can never be completely the same.
+
+fig, ax = plt.subplots(nrows=1)
+im = ax.imshow(
+    np.squeeze(rml.icube.sky_cube.detach().numpy()),
+    origin="lower",
+    interpolation="none",
+    extent=rml.icube.coords.img_ext,
+)
+plt.colorbar(im)
+
+rml.state_dict()
+
+# ### Loading the Model
+#
+# To demonstrate the saving and loading of the model here we will be resetting the parameters of the model and then reloading them.
+
+rml = precomposed.SimpleNet(coords=coords)
+rml.state_dict()  # parameters of the model
+
+# Here you can clearly see the state_dict() returning to its original state from the beginning of the tutorial and then once we reload it, it reverts to the values it was when we saved it.
+
+rml.load_state_dict(torch.load("dirty_image_model.pt"))
+rml.state_dict()  # parameters of the model
+
+# To continue and optimize our model against the visibility data the gridded visibilitied need to have uniform weighting
+
+gridder._grid_visibilities(weighting="uniform")
+img, beam = gridder.get_dirty_image(weighting="uniform")
+dataset = gridder.to_pytorch_dataset()  # creating a new dataset
+
+# ### Conclusion
+#
+# This tutorial shows how to work towards a better starting point for RML optimization by using an image with maximal likelihood but RML optimization does not need to start from here and can start from a blank image. This is unlike a deconvolution algorithm such as the CASA tclean. The training loop done here is just the set up and the model/image still needs further regularization and optimization against the visibility data.
```

Comment:
You can link to the optimization and cross validation tutorials for more information